### ISSUE TITLE
Support for CQL array predicates

### ISSUE THREAD

**Issue by cportele** [interactive-instruments/ldproxy#431]
Note: This is a follow-up issue from #395.

CQL has recently added support for [array predicates](https://docs.ogc.org/DRAFTS/19-079r1.html#arrays) and now comparisons are not restricted to single values (string, number, boolean, time instant, time interval, spatial geometry), but also sets of them.

Currently, in ldproxy all queryables are treated as a single value. This is also the case for properties that are an array or that are embedded in an array of objects. 

In the current implementation we can use the predicates to select the features where at least one of the values in the (potentially nested) array meets the predicate. 

An example from the [codelist records of the GeoInfoDok](https://demo.ldproxy.net/geoinfodok/collections/ax_gebaeudefunktion/items?f=json):

`theme.concept` is a queryable and both `theme` and `concept` are array values. The value of theme is an array of objects, and `concept` is a member of each object and is an array of strings:

```json
{
  "type": "Feature",
  "id": "1000",
  "geometry": null,
  "properties": {
    "type": "Werteart",
    "title": "Wohngebäude",
    "description": "'Wohngebäude' ist ein Gebäude, das zum Wohnen genutzt wird.",
    "language": "de",
    "created": "2019-06-01T00:00:00",
    "updated": "2019-06-01T00:00:00",
    "theme": [
      {
        "scheme": "profile",
        "concept": [
          "DLKM",
          "Basis-DLM",
          "DLM50",
          "DLKM (Grunddatenbestand)"
        ]
      }
    ]
  }
}
```

For example, the following selects all values that have in an embedded `theme.concept` array a value "DLM50":

https://demo.ldproxy.net/geoinfodok/collections/ax_gebaeudefunktion/items?f=json&filter=theme.concept=%27DLM50%27

Consistent with this we currently specify the queryable as a string-valued property.

The array predicates (equal, contains/contained by, overlaps) would also allow to, for example, select all records where `concept` contains "DLKM", "BasisDLM" and "DLM50". Currently this is not possible.

This, however, would require that we change the queryable model.

In simpler cases with no arrays of objects this is works well. If we just consider the `concept` property, it is already a VALUE_ARRAY (string) in the provider schema.

Where currently requests like the following are created

```
SELECT AA.id FROM record AA JOIN record_scheme AB ON (AA.id=AB.record_fk) JOIN record_scheme_concept AC ON (AB.id=AC.scheme_fk) WHERE AC.concept = 'DLM50')
```

the array representation would be 

```
SELECT AA.id FROM record AA JOIN record_scheme AB ON (AA.id=AB.record_fk) where ARRAY(SELECT AC.concept FROM record_scheme_concept AC WHERE AC.scheme_fk=AB.id) @> ARRAY['DLM50']
```

This uses the PostgreSQL ARRAY concept. A better approach would be to use standard SQL so that the queries would also work in other RDBMS like GeoPackage/SQLite.

The situation gets more complicated in schemas that include OBJECT_ARRAY properties like `theme`. Neither CQL nor OGC API Features provide a guidance for such cases, so this will require an ldproxy-specific solution.

Since the concept of nested filters is already supported as a CQL extension for property references, the best choice seems to be

- to continue to "ignore" intermediate OBJECT_ARRAY properties when determining the type of a queryable;
- to treat all VALUE_ARRAY properties as array properties in CQL;
- to support sub-filters on intermediate OBJECT_ARRAY properties.

With this the `theme.concept` queryable would be specified as 

```json
"theme.concept": {
  "type": "array",
  "items": {
	 "type": "string"
  } 
}
```

and a filter that selects all records where `concept` contains "DLKM", "Basis-DLM" and "DLM50" in a "profile" theme would be written as:

```
theme[scheme = 'profile'].concept ACONTAINS ['DLKM', 'Basis-DLM', 'DLM50']
```

Related to https://github.com/interactive-instruments/ldproxy-project-management/issues/43

**Comment by azahnen** [comment]
@bekakh The configuration is here: https://gitea.ci.interactive-instruments.de/ldproxy/demo/src/branch/master/store/entities/providers/geoinfodok.yml

**Comment by cportele** [comment]
addressed by #490
